src/xy.js

```diff
@@ -17,9 +15,15 @@ var XYControl = React.createClass({
   },
   getInitialState() {
     return {
-      dragging: false
+      dragging: false,
+      isMobile: false
     };
   },
+  componentDidMount() {
+    this.setState({
+      isMobile: typeof document != 'undefined' && 'ontouchstart' in document
```

Comment:
That's true. The reason was to encapsulate the value in the component, and only perform validation on the client side on componentDidMount.
But it seems a good idea leave the previous design instead.
Let's reject the pull request and I'm going to perform a new modification over code.